apps/jobs/views.py

```diff
@@ -229,3 +231,72 @@ def leaderboard(request, challenge_phase_split_id):
     paginator, result_page = paginated_queryset(distinct_sorted_leaderboard_data, request)
     response_data = result_page
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def remaining_submission(request, challenge_phase_id, participant_team_id, challenge_id):
+    try:
+        Challenge.objects.get(pk=challenge_id)
+    except Challenge.DoesNotExist:
+        response_data = {'error': 'Challenge does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        ChallengePhase.objects.get(pk=challenge_phase_id)
+    except ChallengePhase.DoesNotExist:
+        response_data = {'error': 'Challenge Phase does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        ParticipantTeam.objects.get(pk=participant_team_id)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {'error': 'Participant team does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    participant_team_id = get_participant_team_id_of_user_for_a_challenge(
+        request.user, challenge_id)
+
+    try:
+        ParticipantTeam.objects.get(pk=participant_team_id)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {'error': 'You haven\'t participated in the challenge'}
+        return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    max_submission_per_day = ChallengePhase.objects.get(
+            pk=challenge_phase_id).max_submissions_per_day
+
+    max_submission = ChallengePhase.objects.get(
+            pk=challenge_phase_id).max_submissions
+
+    submissions_done_today_count = Submission.objects.filter(
+        challenge_phase__challenge=challenge_id,
+        participant_team=participant_team_id,
+        challenge_phase=challenge_phase_id,
+        submitted_at__gte=datetime.date.today()).count()
+
+    failed_count = Submission.objects.filter(
+        challenge_phase=challenge_phase_id,
+        participant_team=participant_team_id,
+        challenge_phase__challenge=challenge_id,
+        status=Submission.FAILED,
+        submitted_at__gte=datetime.date.today()).count()
+
+    if (submissions_done_today_count - failed_count) >= max_submission_per_day or max_submission_per_day == 0:
+        dtnow = datetime.datetime.now()
+        dttomorrow = dtnow + datetime.timedelta(days=1)
+        midnight = dttomorrow.replace(hour=0, minute=0, second=0)
+        remaining_time = midnight - dtnow
```

Comment:
A better way of calculating would be 

```python
from django.utils import timezone
from datetime import timedelta, datetime, time

now = timezone.now()
date_tomorrow = today.date() + timedelta(1)
midnight =  datetime.combine(date_tomorrow, time())
time_left  = midnight - now
# Finally user time_left accordingly in your code 
```

This will make sure that you are using the `timezone` module that is Timezone Aware.